expr_test.go

```diff
@@ -1736,6 +1741,28 @@ func TestFunction(t *testing.T) {
 	assert.Equal(t, 20, out)
 }
 
+func TestFunctionWithContext(t *testing.T) {
+	add := expr.FunctionWithContext(
+		"add",
+		func(p ...interface{}) (interface{}, error) {
+			out := 0
+			fmt.Println("p", len(p))
+			for _, each := range p[1:] {
```

Comment:
Let's also add the test of what actually uses the context feature. Something with terminating.